code/model/TimedNotice.php

```diff
@@ -178,6 +184,60 @@ public function canCreate($member = null)
     }
 
     /**
+     * Gets any notices relevant to the present time, context and current users
+     *
+     * @return ArrayList
+     **/
+    public static function get_notices()
+    {
+        // analyse the context this message gets requested
+        $context = self::getContext();
```

Comment:
Ah yeap, I can understand that. Perhaps a solution then could be to make
the context parameter default to CMS. It's more explicit and would also
allow developers using the notices on their website frontend to make use
the TimedNoticeController, giving them the snoozing functionality.



On Sat, Feb 20, 2016 at 6:28 PM, Peter Thaleikis <notifications@github.com>
wrote:

> In code/model/TimedNotice.php
> <https://github.com/sheadawson/silverstripe-timednotices/pull/14#discussion_r53544958>
> :
>
> > @@ -178,6 +184,60 @@ public function canCreate($member = null)
> >      }
> >
> >      /**
> > +     * Gets any notices relevant to the present time, context and current users
> > +     *
> > +     * @return ArrayList
> > +     **/
> > +    public static function get_notices()
> > +    {
> > +        // analyse the context this message gets requested
> > +        $context = self::getContext();
>
> (facepalm) I should have picked this up.
>
> About the context passing in: I intentionally didn't pass it in because
> this would give developers the chance to set it accidentally to 'CMS' while
> actually on the website and exposing CMS-related messages to a potential
> attacker. What do you think?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sheadawson/silverstripe-timednotices/pull/14/files#r53544958>
> .
>



-- 
*Shea Dawson*
Web Developer
LiveSource

*mobile* +64 20 4028 5647
*skype* squatchnz
*email* shea@livesource.co.nz
*web *www.livesource.co.nz